mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java

```diff
@@ -152,11 +153,13 @@
         private final ComposedMessageId id;
         private final Long size;
         private final Optional<List<MessageAttachmentMetadata>> messageAttachments;
+        private final ThreadId threadId;
 
         public AppendResult(ComposedMessageId id, Long size, Optional<List<MessageAttachmentMetadata>> messageAttachments) {
             this.id = id;
             this.size = size;
             this.messageAttachments = messageAttachments;
+            this.threadId = new ThreadId(id.getMessageId());
```

Comment:
This feels wrong.

1. We can have a factory method for ThreadId to make things more explicit:

```
this.threadId = ThreadId.forBaseMessageId(id.getMessageId());
```

2. We are leaking some business decisions all around our data model.

`new ThreadId(id.getMessageId())` is a decision that I would love to see taken only once.

This means we need to cary over the threadId along in the data model. We need to

 - Have a constructor parameter for `threadId` here in `AppendResult`
  - Cary the threadId parameter over in `MessageMetaData` POJO.